### ISSUE TITLE
Fatal error during worker execution (getShareOpeningTag called with NULL argument)

### ISSUE THREAD

**Issue by gouttegd** [friendica/friendica#8874]
### Expected behavior

The `worker.php` script executes without any error.

### Actual behavior

The `worker.php` script runs into a PHP Fatal error:

```
PHP Fatal error:  Uncaught TypeError: Argument 4 passed to Friendica\Content\Text\BBCode::getShareOpeningTag() must be of the type string, null given, called in /var/www/social.incenp.org/src/Protocol/ActivityPub/Transmitter.php on line 867 and defined in /var/www/social.incenp.org/src/Content/Text/BBCode.php:2226
Stack trace:
#0 /var/www/social.incenp.org/src/Protocol/ActivityPub/Transmitter.php(867): Friendica\Content\Text\BBCode::getShareOpeningTag('Damien Goutte-G...', 'https://social....', 'https://social....', NULL, '2015-12-30 11:3...', '6e391d55375683c...')
#1 /var/www/social.incenp.org/src/Protocol/ActivityPub/Transmitter.php(837): Friendica\Protocol\ActivityPub\Transmitter::createActivityFromItem(29325)
#2 /var/www/social.incenp.org/src/Worker/Notifier.php(802): Friendica\Protocol\ActivityPub\Transmitter::createCachedActivityFromItem(29325, true)
#3 /var/www/social.incenp.org/src/Worker/Notifier.php(179): Friendica\Worker\Notifier::activityPubDelivery('uplink', Array, Array, Array, 50, '2020-07-12 21:0...', Array)
#4 in /var/www/social.incenp.org/src/Content/Text/BBCode.php on line 2226
```

It looks somewhat similar to #8866, except that it’s not fixed by #8867.

### Steps to reproduce the problem

Unknown, didn’t try to reproduce on a fresh installation.

### Friendica version you encountered the problem

Friendica 2020.07. Never seen that problem before upgrading to that version.

### Friendica source (git, zip)

Git

### PHP version

PHP 7.3.19

### SQL version

MariaDB 10.3.22

**Comment by MrPetovan** [comment]
Thank you for the report!

**Comment by gouttegd** [comment]
The 4th argument to `Friendica\Content\Text\BBCode::getShareOpeningTag()` in `Friendica\Protocol\ActivityPub\Transmitter::createActivityFromItem()` is supposed to be the item’s *plink*:

```
$item['body'] = BBCode::getShareOpeningTag($item['author-name'], $item['author-link'], $item['author-avatar'], $item['plink'], $item['created'], $item['guid']) . $item['body'] . '[/share]';
```

However in my database, the `plink` column in the `item` table is *always* empty, hence the fatal error. I have no idea whether this is normal or not.

**Comment by MrPetovan** [comment]
This is a little more complex than that. The `plink` that is referred to is supposed to be an attribute of a `[share]` BBCode tag. The `$item` variable is holding all the attributes of this tag and I didn't expect it to be ever missing.

**Comment by gouttegd** [comment]
Well, all I know is that I have 9149 rows in my `item` table, and *none* of them has a value in the `plink` column. If you need any more information let me know, I’ll provide what I can.

**Comment by MrPetovan** [comment]
I retracted my comment, it doesn't have anything to do with `[share]` BBCode tag attributes. I defer to @annando , he worked on moving the `plink` column out of the `item` table, which is why all the values are NULL. The question is where is these values now?

**Comment by annando** [comment]
The values are in the `item-content` table. They are linked via the `uri-id` value.

**Comment by MrPetovan** [comment]
Do we always expect a value?

**Comment by annando** [comment]
I had to check in the code.

**Comment by MrPetovan** [comment]
This `getShareOpeningTag()` call has been removed in bf7128b4b2550204c503baa025ddb746f54d444b and as such this issue has been fixed by #9001
